### PR TITLE
WIP: ci: update pipeline

### PR DESCRIPTION
**Pull request by jsmassa**
1) Ports from leiningen to deps.edn
2) Enables use of pipeline as in other projects

Notes: 
- Format check will be enabled in follow-up PR

Can we simply set this as default branch?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,10 @@ pom.xml.asc
 /.nrepl-port
 .hgignore
 .hg/
+.cache
+.cpcache
+node_modules/
+package-lock.json
+package.json
```
Comment (TimoKramer) [inline .gitignore:16]: these two files should not be ignored

File: build.clj [hunk 2]
```diff
@@ -0,0 +1,76 @@
+(ns build
+  (:refer-clojure :exclude [test compile])
+  (:require [clojure.tools.build.api :as b]
+            [borkdude.gh-release-artifact :as gh]
+            [org.corfield.build :as bb])
```
Comment (TimoKramer) [inline build.clj]: this is deprecated unfortunately, just copy the build.clj from datahike-s3 or others

File: deps.edn [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{:deps {org.clojure/clojure {:mvn/version "1.11.1"}}
+ :aliases {:test {:main-opts ["-m" "kaocha.runner"]
+                  :extra-deps {org.clojure/clojurescript {:mvn/version "1.11.60"}
+                               lambdaisland/kaocha {:mvn/version "1.71.1119"}
+                               lambdaisland/kaocha-cljs {:mvn/version "1.4.130"}}
+                  :extra-paths ["test"]}
+           :format {:extra-deps {cljfmt/cljfmt {:mvn/version "0.8.0"}}
+                    :main-opts ["-m" "cljfmt.main" "check"]}
+           :build {:deps {io.github.seancorfield/build-clj {:git/tag "v0.8.2"
+                                                            :git/sha "0ffdb4c"}
```
Comment (TimoKramer) [inline deps.edn]: same, just copy from datahike-s3, konserve, hasch or incognito

**Review by TimoKramer — APPROVED** [review]
in settings you can just set to default-branch main 👍

File: .gitignore (hunk 1 above)
Comment (jsmassa) [reply .gitignore:16]: They are being created automatically though, so I don't see a reason to have them in the repo

File: build.clj (hunk 2 above)
Comment (jsmassa) [reply build.clj]: Ah, then the datalog-parser probably needs an update as well as this is where I copied the stuff from

File: .gitignore (hunk 1 above)
Comment (TimoKramer) [reply .gitignore:16]: hmm, I wouldn't ignore them and just commit them but do as you prefer

**Review by TimoKramer — APPROVED** [review]
I think the template/pom.xml needs to be created as well

**Comment by jsmassa** [comment]
The template/pom.xml had me confused for a bit as I thought I added that. Turns out there was a 'pom.xml' in the .gitignore...

File: build.clj (hunk 2 above)
Comment (jsmassa) [reply build.clj]: Done.

File: deps.edn (hunk 3 above)
Comment (jsmassa) [reply deps.edn]: Done.

**Review by TimoKramer — APPROVED** [review]

**Pull request closed without merging**